vllm/engine/metrics.py

```diff
@@ -1,51 +1,151 @@
-from aioprometheus import Gauge
+from vllm.logger import init_logger
+from aioprometheus import Counter, Gauge, Histogram
+
+import time
+import numpy as np
+from typing import List
+from dataclasses import dataclass
+
+logger = init_logger(__name__)
+
+labels = {}
+
+
+def add_global_metrics_labels(**kwargs):
+    labels.update(kwargs)
+
 
 # The begin-* and end* here are used by the documentation generator
 # to extract the metrics definitions.
 
 # begin-metrics-definitions
-gauge_avg_prompt_throughput = Gauge("vllm:avg_prompt_throughput_toks_per_s",
-                                    "Average prefill throughput in tokens/s.")
-gauge_avg_generation_throughput = Gauge(
-    "vllm:avg_generation_throughput_toks_per_s",
-    "Average generation throughput in tokens/s.")
+counter_prompt_tokens = Counter("vllm_prompt_tokens_total",
+                                "Number of prefill tokens processed.")
+counter_generation_tokens = Counter("vllm_generation_tokens_total",
+                                    "Number of generation tokens processed.")
 
 gauge_scheduler_running = Gauge(
-    "vllm:num_requests_running",
-    "Number of requests that is currently running for inference.")
-gauge_scheduler_swapped = Gauge("vllm:num_requests_swapped",
-                                "Number requests swapped to CPU.")
-gauge_scheduler_waiting = Gauge("vllm:num_requests_waiting",
+    "vllm_requests_running_total",
+    "Number of requests currently running on GPU.")
+gauge_scheduler_swapped = Gauge("vllm_requests_swapped_total",
+                                "Number of requests swapped to CPU.")
+gauge_scheduler_waiting = Gauge("vllm_requests_waiting_total",
                                 "Number of requests waiting to be processed.")
 
 gauge_gpu_cache_usage = Gauge(
-    "vllm:gpu_cache_usage_perc",
+    "vllm_gpu_cache_usage_perc",
     "GPU KV-cache usage. 1 means 100 percent usage.")
 gauge_cpu_cache_usage = Gauge(
-    "vllm:cpu_cache_usage_perc",
+    "vllm_cpu_cache_usage_perc",
     "CPU KV-cache usage. 1 means 100 percent usage.")
+
+histogram_time_to_first_token = Histogram(
+    "vllm_time_to_first_token_seconds",
+    "Histogram of time to first token in seconds.",
+    buckets=[0.05, 0.1, 0.25, 0.5, 0.75, 1.0, 2.5, 5.0, 7.5, 10.0])
```

Comment:
For the default we probably need more lower bounds because a single prefill can take less than 50ms.  how about 0.001, 0.005, 0.01, 0.02, 0.04, 0.06, 0.08, 0.1, ...

Reply:
Sure, will update. Ultimately we should have a way for the user to configure this. 

I think something we should consider as a feature is making YAML config files for the server so people don't have to pass in so many arguments. As part of this, we could let the user specify buckets for the logging.